### DISCUSSION TITLE
Add queue/jobs support

### DISCUSSION THREAD

**Discussion by MekDrop** [ImpressCMS/impresscms#1028] [Ideas]
We need to have some functionality to be able to write big tasks that can be written simpler and could be divided between machines. This functionality partially will replace autotasks functionality.

There are two candidates that possible can be used for that `bernard/bernard` and `enqueue/enqueue`.

**Comment by MekDrop** [comment]
I found third candidate and I think I'll try with it - `swarrot/swarrot`. Chained processors for me looks powerful but also easy enough to understand. We will see if this way would work.

**Comment by fiammybe** [comment]
Will there be a possibility to get this kind of functionality working with a shared-hosting account? That remains our baseline config for now.

**Comment by MekDrop** [comment]
At the moment on shared hosting that will be achieved with file driver. Later (after we solve ImpressCMS/impresscms#160 ticket) MySQL support will be included. 

Other possible options are listed in [imponeer/queue-interop-connection-factory-helper](https://github.com/imponeer/queue-interop-connection-factory-helper). Maybe some of them also could be used on shared hosting. I haven't investigated them yet. Also, some more other options also could be included in the future.
